xds/src/main/java/io/grpc/xds/XdsNameResolver.java

```diff
@@ -127,6 +128,7 @@ final class XdsNameResolver extends NameResolver {
   private final long randomChannelId;
 
   private volatile RoutingConfig routingConfig = RoutingConfig.empty;
+  private volatile MetricRecorder metricRecorder;
```

Comment:
Why is this volatile? It is set once during construction. Seems it could even be `final`.

Reply:
That was an oversight, updated.